src/builder/edit_automod_rule.rs

```diff
@@ -0,0 +1,122 @@
+use std::collections::HashMap;
+
+use crate::json::{json, Value};
+use crate::model::guild::automod::{Action, EventType, Trigger};
+use crate::model::id::{ChannelId, RoleId};
+
+#[derive(Debug, Default, Clone)]
+pub struct EditAutoModRule(pub HashMap<&'static str, Value>);
+
+impl EditAutoModRule {
+    /// The display name of the rule.
+    pub fn name<D: ToString>(&mut self, name: D) -> &mut Self {
+        self.0.insert("name", Value::from(name.to_string()));
+
+        self
+    }
+
+    /// Set the event context the rule should be checked.
+    pub fn event_type(&mut self, event_type: EventType) -> &mut Self {
+        self.0.insert("event_type", u8::from(event_type).into());
+
+        self
+    }
+
+    /// Set the type of content which can trigger the rule.
+    ///
+    /// **None**: The trigger type can't be edited. Only its values.
```

Comment:
EditAutoModRule is almost a full copy of CreateAutoModRole. Maybe the same type can be used for both? With basically just a small change here: "\*\*No**t**e\*\*: **When editing a rule,** the trigger type can't be edited. Only its values."

Reply:
That can probably said about every Create/Edit builder combination in serenity.

`EditAutoModRule` doesn't set the `trigger_type`, only the metadata.

Reply:
Alright I guess then let's keep the duplication for consistency lol

Reply:
Ok, I had the impression the builders come in pairs (create/edit) but it's actually either `Create*` or `Edit*`. 
I was not looking for a pattern of when to use one or the other.
I rolled a dice and `Edit` stays.

Reply:
I haven't looked at this closely, but I will say there's a precedent here. Specifically, `CreateInteractionResponseFollowup` is used for both creating and editing followup interaction messages. I don't know if there's any other builders that are used for both.

Reply:
Should probably rename to `CreateAutoModRole` then for consistency

Reply:
Found a counter-example in `EditRole`, which is used for both creation and editing of roles.